src/cljs/rems/administration/blacklist.cljs

```diff
@@ -26,25 +27,100 @@
              :error-handler (flash-message/default-error-handler :top description)}))
    {:db (assoc db ::loading? true)}))
 
+(rf/reg-event-fx
+ ::add-to-blacklist
+ (fn [{:keys [db]} [_ resource user comment]]
+   (let [description [text :t.administration/add]]
+     (post! "/api/blacklist/add"
+            {:params {:blacklist/resource (select-keys resource [:resource/ext-id])
+                      :blacklist/user (select-keys user [:userid])
+                      :comment comment}
+             :handler (flash-message/default-success-handler
+                       :top
+                       description
+                       (fn []
+                         (rf/dispatch [::fetch-blacklist])
+                         (rf/dispatch [::set-selected-user nil])))
+             :error-handler (flash-message/default-error-handler :top description)}))
+   {}))
+
 (rf/reg-event-fx
  ::remove-from-blacklist
- (fn [{:keys [db]} [_ resource user]]
+ (fn [{:keys [db]} [_ resource user comment]]
    (let [description [text :t.administration/remove]]
      (post! "/api/blacklist/remove"
             {:params {:blacklist/resource (select-keys resource [:resource/ext-id])
                       :blacklist/user (select-keys user [:userid])
-                      :comment ""}
+                      :comment comment}
              :handler (flash-message/default-success-handler
                        :top
                        description
-                       #(rf/dispatch [::fetch]))
+                       #(rf/dispatch [::fetch-blacklist]))
              :error-handler (flash-message/default-error-handler :top description)}))
    {}))
 
+(rf/reg-event-fx
+ ::fetch-users
+ (fn []
+   (fetch "/api/blacklist/users"
+          {:handler #(rf/dispatch [::fetch-users-result %])
+           :error-handler (flash-message/default-error-handler :top "Fetch users")})
+   {}))
+
+(rf/reg-event-db
+ ::fetch-users-result
+ (fn [db [_ users]]
+   (assoc db ::all-users (map atoms/enrich-user users))))
+
+(rf/reg-sub
+ ::all-users
+ (fn [db _]
+   (::all-users db)))
+
+(rf/reg-event-db
+ ::set-selected-user
+ (fn [db [_ user]]
+   (assoc db ::selected-user user)))
+
+(rf/reg-sub
+ ::selected-user
+ (fn [db _]
+   (::selected-user db)))
+
+(defn add-user-form [resource]
+  (let [user-field-id "blacklist-user"
+        all-users @(rf/subscribe [::all-users])
+        selected-users @(rf/subscribe [::selected-user])]
+    [:form.form-inline
+     {:on-submit (fn [event]
```

Comment:
we haven't used on-submit callbacks so far, but instead on-click callbacks on buttons. any reason for an on-submit here?

Reply:
No big reason. That way it submits also by pressing enter in the user field.